tests/fs/test_getdents64_special_cases.cpp

```diff
@@ -51,6 +54,8 @@ int main()
     // Non-ascii file name.
     doTest("test_dir", u8"абвгд");
 
+#ifndef WASMFS // The JS FS truncates filenames automatically, which is incorrect. Wasmfs and Linux do not.
```

Comment:
Seems sad not to fix the JS FS and update the test so it passes correctly everywhere..

Reply:
I agree that would be optimal, but the tradeoff is spending time on the old system we are working to replace. I looked before and didn't see a quick fix, but maybe you're right and we should try more. I looked into it now and it seems like the old FS has no size limit: it simply allows any size for filenames (I guess limited by JS string limits). The truncation of the filename happens in `getdents` actually, as it writes out the names:

https://github.com/emscripten-core/emscripten/blob/9b15472c2c1e43cd19278be95c88de36f07bb770/src/library_syscall.js#L702

Apparently [POSIX limits filenames to 255](https://www.gnu.org/software/automake/manual/html_node/Limitations-on-File-Names.html), so the old FS is exceeding POSIX. But I'm not sure if it's worth refactoring the old FS to change this or not?

Reply:
Sure.. lgtm